app/src/main/kotlin/com/flxrs/dankchat/data/NotificationService.kt

```diff
@@ -254,10 +263,22 @@ class NotificationService : Service(), CoroutineScope {
             ttsMessageQueue -> TextToSpeech.QUEUE_ADD
             else            -> TextToSpeech.QUEUE_FLUSH
         }
-        val ttsMessage = when (name) {
+        var ttsMessage = when (name) {
             previousTTSUser -> message
             else            -> messageFormat.also { previousTTSUser = name }
         }
+        if (removeURL) {
+            val urlPattern = Pattern.compile("((https?|ftp|gopher|telnet|file|Unsure|http):((//)|(\\\\))+[\\w\\d:#@%/;$()~_?\\+-=\\\\\\.&]*)", Pattern.CASE_INSENSITIVE)
+            ttsMessage = ttsMessage.replace(urlPattern.toRegex(), "").trim()
+        }
+        if (removeEmote) {
+            for (emote in this.emotes) {
+                if (ttsMessage.contains(emote.code, ignoreCase = true)) {
+                    ttsMessage = ttsMessage.replace(emote.code, "", ignoreCase = true)
+                }
+            }
+            ttsMessage = ttsMessage.replace(Regex("(\\u00a9|\\u00ae|[\\u2000-\\u3300]|\\ud83c[\\ud000-\\udfff]|\\ud83d[\\ud000-\\udfff]|\\ud83e[\\ud000-\\udfff])"), "")
```

Comment:
A comment about what special characters are being replaced here would be nice

Reply:
Fixed in [0c2332a](https://github.com/flex3r/DankChat/pull/309/commits/0c2332a356e6c2dfd44c79514488ca7dfdf67aba)